### DISCUSSION TITLE
How to apply color scale by field range?

### DISCUSSION THREAD

**Discussion by aborruso** [vega/altair#2916] [Q&A] [answered]
In vega-lite I'm able to apply color scale by field range, I'm able to use the field values of `color` field to color my chart ([this is the sample code][1]).

I'm not able to rewrite in Python/altair, the same of this JSON.

```json
    "color": {
      "type": "nominal",
      "field": "label",
      "scale": {"range": {"field": "color"}},
      "title": ""
    }
```

It's not this

```python
chart=alt.Chart(df).mark_rect().encode(
    x='x:O',
    y='y:O',
    color=alt.Color('label:N',scale=range('color:N'))
)
```

  [1]: https://vega.github.io/editor/#/url/vega-lite/N4Igxg9gdgZglgcxALlANzgUwO4tJKAFzigFcJSBnAdTgBNCALFAZgAY2AacaYsiygAlMiRoVYcAvpO50AhoTl4QUOQFtMKEPMUBaABwwAbACYArHMwB2K9bYsTAFgCMRtgE4Wcx1f0AjOj8bMBg6EBkQNTkAJwBrLWjMMHFuTChIOhIkVBAAD2V4TAAbMOQ8kG5CAE8AB00yiGjM1SLw7iqCrBKtDsra+pBG5rlWiMgixuVquq0oCDUSEYqQQu6york-YuXKMBH60Gi5KAQDla7Sngno8IjiQiKB2+4AEl3GTCitMUIaymQAPQAtCYBByAB0CDgTFIfnBcAgAPenzkwNBcl0RWhmGBZnBJnBbHBACtKNBljo5JRMIRKMpKQZjOZLDY7A4XG5PN5fAEglYQqUANqgfLIZztFDiq6TMoAYlCfmMMGWIxqjCUyHMZm4Gy2rTKADloLoAGKNTCUFIgNAjUj1NgyEWSiWa7jjGUgWUsLxsExgFVFNUarU6zbbMpmxKW5Y2op2yWO8pil0sN0Qa5aeUwbPZgNBlBcEC68MgcyOGO2+plyQAXWkQA

**Comment by ChristopherDavisUCI** [comment, marked as answer]
If you ever want to compare your Altair code to the Vega-Lite code, a good trick is to evaluate `print(chart.to_json())`, which can sometimes help with spotting the problem (maybe not as much in this particular example).

I believe this will work: 
```
color=alt.Color('label:N',scale=alt.Scale(range={"field": "color"}))
```
@mattijn Do you see a nicer way than explicitly specifying this dictionary, `{"field": "color"}`?

In a soon-to-be-released version of Altair, you can skip the `alt.Scale` portion:
```
color=alt.Color('label:N').scale(range={"field": "color"})
```

There is also an option to use the raw color values directly, but I couldn't figure out a nice way to also get your labels to show up with this approach: https://altair-viz.github.io/user_guide/customization.html#raw-color-values
```
alt.Chart(df).mark_rect().encode(
    x='x:O',
    y='y:O',
    color=alt.Color('color:N', scale=None)
)
```

**Reply by aborruso** [reply]
> If you ever want to compare your Altair code to the Vega-Lite code, a good trick is to evaluate `print(chart.to_json())`, which can sometimes help with spotting the problem (maybe not as much in this particular example).

I know, but here I had the right vega code, and so I needed a function that did the opposite (`print(chart.to_python())`).

```
color=alt.Color('label:N',scale=alt.Scale(range={"field": "color"}))
```

It works, thank you very much
